cloudfront.tf

```diff
@@ -0,0 +1,102 @@
+locals {
+  s3_origin_id = "robojackets-apiary-maintenance-page"
+}
+
+resource "aws_s3_bucket" "bucket" {
+  bucket_prefix = "gatech-robojackets-apiary-mtc"
+  acl           = "private"
+}
+
+resource "aws_s3_bucket_policy" "bucket_policy" {
+  bucket = aws_s3_bucket.bucket.id
+  policy = data.aws_iam_policy_document.s3_policy.json
+}
+
+resource "aws_cloudfront_origin_access_identity" "oai" {
+  comment = "Apiary maintenance page"
+}
+
+resource "aws_s3_bucket_public_access_block" "block_public_access" {
+  bucket = aws_s3_bucket.bucket.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+data "aws_iam_policy_document" "s3_policy" {
+  statement {
+    actions   = ["s3:GetObject"]
+    resources = ["${aws_s3_bucket.bucket.arn}/*"]
+
+    principals {
+      type        = "AWS"
+      identifiers = [aws_cloudfront_origin_access_identity.oai.iam_arn]
+    }
+  }
+  statement {
+    actions   = ["s3:ListBucket"]
+    resources = [aws_s3_bucket.bucket.arn]
+
+    principals {
+      type        = "AWS"
+      identifiers = [aws_cloudfront_origin_access_identity.oai.iam_arn]
+    }
+  }
+}
+
+resource "aws_cloudfront_distribution" "cloudfront" {
+  aliases = [var.domain_name]
+
+  origin {
+    domain_name = aws_s3_bucket.bucket.bucket_regional_domain_name
+    origin_id   = local.s3_origin_id
+
+    s3_origin_config {
+      origin_access_identity = aws_cloudfront_origin_access_identity.oai.cloudfront_access_identity_path
+    }
+  }
+
+  default_cache_behavior {
+    allowed_methods        = ["HEAD", "GET", "OPTIONS"]
+    cached_methods         = ["HEAD", "GET"]
+    compress               = true
+    viewer_protocol_policy = "redirect-to-https"
+    target_origin_id       = local.s3_origin_id
+
+    forwarded_values {
+      query_string = false
+
+      cookies {
+        forward = "none"
+      }
+    }
+  }
+
+  custom_error_response {
+    error_caching_min_ttl = 60
+    error_code            = 404
+    response_code         = 503
+    response_page_path    = "/maintenance-page.html"
+  }
+
+  default_root_object = "index.html" # this is deliberately a nonexistent object so cloudfront returns a 503
+  enabled             = true
+  is_ipv6_enabled     = false
+  price_class         = "PriceClass_100"
+
+  restrictions {
+    geo_restriction {
+      locations        = ["US"]
```

Comment:
Is there a reason for this restriction?  There are times (study abroad, international student at home, etc.) where RJ members might try to access MyRoboJackets from outside the US

Reply:
It's mostly intended as a cost-saving measure as _most_ clients are in the US. I can take it out.

Do you also want to change the price class? [[docs](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/PriceClass.html)]

Reply:
It sounds like PriceClass_100 without the geographic restriction (if I'm understanding correctly) would just cause higher latency outside the US and Europe and that seems fine to me

Reply:
That's correct.